### ISSUE TITLE
Definition list - content not using semantic markup (1.3.1)

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux-layout#16]
### Expected behavior

- Headings use heading markup (i.e., H2, H3, etc.)
- List, properties, and values use description list markup (dl, dt, dd) - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

Screen reader users have access to headings to use for navigation landmarks and understanding content structure. Screen reader users have access to understand content is a list, how many items are in it, and the relationship between property and value. 

### Actual behavior
Everything is a div.
